src/tests/ftest/vmd/fault_reintegration.py

```diff
@@ -0,0 +1,231 @@
+            dev_state (str): expect dev_state, default to "NORMAL".
+            led_state (str): expect led_state, default to "OFF".
+
+        Returns:
+            bool: True if the expected dev_state and led_state.
+
+        """
+        return self.check_result(self.get_led_status_value(device), dev_state, led_state)
+
+    def check_result(self, result, dev_state, led_state):
+        """Check for result of storage device and led states.
+
+        Args:
+            result (dict): return from get_led_status_value for a storage device.
+            dev_state (str): expect dev_state, default to "NORMAL".
+            led_state (str): expect led_state, default to "OFF".
+
+        Return:
+            True if the expected dev_state and led_state.
+
+        """
+        if "response" in result.keys():
+            for value in list(result['response']['host_storage_map'].values()):
+                if value['storage']['smd_info']['devices']:
+                    for device in value['storage']['smd_info']['devices']:
+                        if device['dev_state'] == dev_state and device['led_state'] == led_state:
+                            return True
+        elif "host_storage_map" in result.keys():
+            for value in list(result['host_storage_map'].values()):
+                if value['storage']['smd_info']['devices']:
+                    for device in value['storage']['smd_info']['devices']:
+                        if device['dev_state'] == dev_state and device['led_state'] == led_state:
+                            return True
+        else:
+            self.log.info("##Unsupported result dictionary %s", result)
+        return False
+
+    def run_ior_with_nvme_fault(self, oclass=None):
+        """Perform ior write with nvme fault testing.
+
+        Test steps:
+            1. Start the DAOS servers.
+            2. Verify the drive status LED.
+            3. Create the pool and container with RF.
+            4. While IOR (EC+GX) in progress put single drive to fault mode.
+            5. IOR will halt until rebuild is finished.
+            6. IOR should complete without any error.
+            7. Drive status LED should be solid Amber ON indicating it’s faulty.
+            8. Replace the same drive back.
+            9. Drive status LED should be off indicating good device is plugged-in.
+           10. Reset and cleanup.
+        Args:
+            oclass (str, optional): object class (eg: RP_2G8, S1,etc). Defaults to None
+        """
+        # 1.
+        self.log_step("Bring up server and reset led of all devices.")
+        if oclass is None:
+            oclass = self.ior_cmd.dfs_oclass.value
+        device_info = get_storage_query_device_info(self, self.dmg)
+        devices = [device['uuid'] for device in device_info]
+        self.log.info("==device_info= %s", devices)
+        test_dev = devices[0]
+        for device in devices:
+            self.run_vmd_led_identify(device, reset=True)
+
+        # 2.
```

Comment:
```suggestion
        # 2. Verify the drive status LED.
```
You might want to add the step description. I always had to scroll up and read the test step in the method description.

Reply:
The step description is on the next line `self.log_step(...)`